lib/credentials/shared_ini_file_credentials.js

```diff
@@ -187,25 +200,33 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   * @param profile [map] credentials profile
   * @throws SharedIniFileCredentialsProviderFailure
   */
-  loadViaCredentialProcess: function loadCredentialProcess(profile, callback) {
-    try {
-      var output = proc.execSync(profile['credential_process']);
-      var credData = JSON.parse(output);
-      if (parseInt(credData.Version) !== 1) {
-        throw AWS.util.error(
-          Error('credential_process does not return Version == 1 for profile ' + this.profile),
-          { code: 'SharedIniFileCredentialsProviderFailure'}
+ loadViaCredentialProcess: function loadViaCredentialProcess(profile, callback) {
+  proc.exec(profile['credential_process'], function(err,stdOut, stdErr) {
+    if (err) {
+      callback(err, null);
+    } else if (stdErr) {
+      callback(stdErr, null);
+    } else {
+      var credData = JSON.parse(stdOut);
+      if (credData.Expiration) {
+        var currentTime = AWS.util.date.getDate();
+        var expireTime = new Date(credData.Expiration);
+        if (expireTime < currentTime) {
+          err = AWS.util.error(
+          new Error('credential_process returned expired credentials'),
+            { code: 'SharedIniFileCredentialsProviderFailure' }
+          );
+        }
+      } else if (parseInt(credData.Version) !== 1) {
```

Comment:
`parseInt` will allow for a string and drop anything after the first character which is not a digit... It probably doesn't matter, but using `parseInt` here does make this implementation more lenient than the docs indicate is allowed.

`parseInt` returns `1` for all of these inputs:
```
1
1.0
1.5
'1what'
```

Reply:
Tightened to only accept 1. Thanks!